File: lyrics_app_android/app/src/main/java/com/maxlisui/lyrics_app/model/SpotifySong.kt
package com.maxlisui.lyrics_app.model

class SpotifySong {
    var isPlaying = false
    var currentlyPlayingType = ""
    var id = ""
    var name = ""
    var album = SpotifyAlbum()
    var artists = mutableListOf<SpotifyArtist>()
}

File: lyrics_app_ios/lyrics_app_ios/Model/SpotifySong.swift
import ObjectMapper

class SpotifySong: Mappable {
    var isPlaying: Bool = false
    var currentlyPlayingType: String = ""
    var id: String = ""
    var name: String = ""
    var album: SpotifyAlbum?
    var artists: [SpotifyArtist] = [SpotifyArtist]()
    
    required init?(map: Map) {
    }
    
    func mapping(map: Map) {
        isPlaying <- map["is_playing"]
        currentlyPlayingType <- map["currently_playing_type"]
        name <- map["item.name"]
        id <- map["item.id"]
        album <- map["item.album"]
        artists <- map["item.artists"]
    }
}
